testsupport/wait/host.go

```diff
@@ -2166,3 +2166,40 @@ func EncodeUserIdentifier(subject string) string {
 
 	return encoded
 }
+
+// CreateSpaceAndSpaceBinding creates a space and spacebindig and waits until both are present.
+// We are creating both of them (Space and SpaceBinding) at the same time , with polling logic, so that we mitigate the issue with spacecleanup_controller deleting the Space before we create it's SpaceBinding.
+func (a *HostAwaitility) CreateSpaceAndSpaceBinding(t *testing.T, mur *toolchainv1alpha1.MasterUserRecord, space *toolchainv1alpha1.Space, spaceRole string, terminatingSpace toolchainv1alpha1.Condition) (*toolchainv1alpha1.Space, *toolchainv1alpha1.SpaceBinding, error) {
+	var spaceBinding *toolchainv1alpha1.SpaceBinding
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		// create the space
+		if err := a.CreateWithCleanup(t, space); err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
```

Comment:
when we create the Space, then I don't think that we could get "not found" error
we should rather check if it already exists
```suggestion
			if !errors.IsAlreadyExists(err) {
				return false, err
			}
```

Reply:
Good catch! fixed in https://github.com/codeready-toolchain/toolchain-e2e/pull/689/commits/6810c96b1fd58e73f4637c474d4a36135a1419c7